app/models/trade.rb

```diff
@@ -0,0 +1,38 @@
+class Trade < ActiveRecord::Base
+  include Referenceable
+  # relationships
+  belongs_to :user
+  belongs_to :partner, class_name: "User"
+  belongs_to :player
+  has_many :offers
+
+  # validations
+  validates_numericality_of :user_rating, only_integer: true, allow_nil: true
+  validates_numericality_of :partner_rating, only_integer: true, allow_nil: true
+  validate -> { reference_exists_in_system("User") }
+  validate -> { reference_exists_in_system("Player") }
+  validate :partner_and_offer_are_valid?, if: :partner_id_changed?
+
+  # we could get pretty detailed with different statuses, but I figured we need
+  # a way to determine trades without a confirmed partner, completed trades to
+  # use to see ratings, and closed (canceled). The other two aren't as important
+  # for people looking for trades, but are intended for people involved with the trade
+  validates_inclusion_of :status, in: ["Open", "Partner Found", "Awaiting Ratings", "Complete", "Closed"]
```

Comment:
Good suggestion. Is this more along the lines of what you're looking for with a yaml file?
in config/trade_statuses.yml have
```
open: "Open"
partnered: "Partner Found"
ratings: "Awaiting Ratings"
complete: "Complete"
closed: "Closed"
```
and use them like this in the Trade model
```
STATUSES = YAML.load_file("#{Rails.root}/config/trade_statuses.yml")
validates_inclusion_of :status, in: STATUSES.values

# scopes
scope :open, -> { where(status: STATUSES["open"]) }
scope :complete, -> { where(status: STATUSES["complete"]) }
```
andreference them anywhere outside of the model with `Trade::STATUSES["open"]` ?